pkg/secrets-store/utils.go

```diff
@@ -0,0 +1,790 @@
+		}
+	} else {
+		if strings.Contains(targetPath, `/var/lib/kubelet/pods`) {
+			parts = strings.Split(targetPath, `/`)
+		}
+	}
+	if len(parts) > 6 {
+		return parts[5]
+	}
+	return ""
+}
+
+// ensureMountPoint ensures mount point is valid
+func (ns *nodeServer) ensureMountPoint(target string) (bool, error) {
+	notMnt, err := ns.mounter.IsLikelyNotMountPoint(target)
+	if err != nil && !os.IsNotExist(err) {
+		return !notMnt, err
+	}
+
+	if !notMnt {
+		// testing original mount point, make sure the mount link is valid
+		_, err := ioutil.ReadDir(target)
+		if err == nil {
+			log.Infof("already mounted to target %s", target)
+			// already mounted
+			return !notMnt, nil
+		}
+		if err := ns.mounter.Unmount(target); err != nil {
+			log.Errorf("Unmount directory %s failed with %v", target, err)
+			return !notMnt, err
+		}
+		notMnt = true
+		// remount it in node publish
+		return !notMnt, err
+	}
+
+	if runtime.GOOS == "windows" {
+		// IsLikelyNotMountPoint always returns notMnt=true for windows as the
+		// target path is not a soft link to the global mount
+		// instead check if the dir exists for windows and if it's not empty
+		// If there are contents in the dir, then objects are already mounted
+		f, err := ioutil.ReadDir(target)
+		if err != nil {
+			return !notMnt, err
+		}
+		if len(f) > 0 {
+			notMnt = false
+			return !notMnt, err
+		}
+	}
+
+	return false, nil
+}
+
+// syncK8sObjects creates or updates K8s secrets based on secretProviderClass spec and data from mounted files in targetPath
+// it should also add pod info to the secretProviderClass object's byPod status field
+func syncK8sObjects(ctx context.Context, targetPath string, podUID string, namespace string, secretProviderClass string, secretObjects []interface{}) error {
+	successfulUpdates := 0
+	files, err := getMountedFiles(targetPath)
+	if err != nil {
+		return err
+	}
+	for _, s := range secretObjects {
+		secretObject, ok := s.(map[string]interface{})
+		if !ok {
+			log.Infof("could not cast secretObject as map[string]interface{} for pod: %s, ns: %s", podUID, namespace)
+			continue
+		}
+		secretName, err := getStringFromObject(secretObject, "secretName")
+		if err != nil {
+			continue
+		}
+		sType, err := getStringFromObject(secretObject, "type")
+		if err != nil {
+			continue
+		}
+		secretType := getSecretType(sType)
+		secretObjectDataList, err := getSliceFromObject(secretObject, "data")
+		if err != nil {
+			continue
+		}
+		datamap := make(map[string][]byte)
+		for _, d := range secretObjectDataList {
+			secretObjectData, ok := d.(map[string]interface{})
+			if !ok {
+				continue
+			}
+			objectName, err := getStringFromObject(secretObjectData, "objectName")
+			if err != nil {
+				continue
+			}
+			key, err := getStringFromObject(secretObjectData, "key")
+			if err != nil {
+				continue
+			}
+			found := false
+			for _, file := range files {
+				filename := filepath.Base(file)
+				if filename == objectName {
```

Comment:
This will fail for objects that have object alias?

Reply:
Yes it will fail. Added #190 as a follow up.